### PR TITLE
Import samples (closes #321)

### LINKED ISSUE #321: As a developer, I would like to import and use sample smart contract projects
As a developer, I want to be able to import sample smart contract projects which demonstrate good practices.

- [ ] I can view the smart contract project README
- [ ] I can import a sample smart contract project
- [ ] I can view any associated applications

### PR DESCRIPTION
**Pull request by Jakeeyturner**
Signed-off-by: Jake Turner <jaketurner25@live.com>

### CODE REVIEW

File: client/src/webview/SampleView.ts [hunk 1]
```diff
@@ -0,0 +1,293 @@
...
+    /**
+     * Opens a file (contract of application)
+     * @param repoName {String} The name of the repository
+     * @param sampleName {String} The name of the sample
+     * @param fileType {String} Either 'contracts' or 'applications'
+     * @param fileName {String} Either the name of the application or the contract
+     * @param language {String} If fileType is 'contracts', then this will be the language of the contract (fileName)
+     * @returns {void}
+     */
+    public static async openFile(repoName: string, sampleName: string, fileType: string, fileName: string, language?: string): Promise<void> {
```
Comment (caroline-church) [inline client/src/webview/SampleView.ts]: can you break this function up its really long

File: client/templates/HomeView.ejs [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+        }
+
+        body.vscode-dark .repository-name {
+            color: #F3F3F3;
+        }
+    </style>
+</head>
+
+<body>
+    <div style="flex-shrink:1;max-width:64px;">
```
Comment (caroline-church) [inline client/templates/HomeView.ejs]: why do you have lots of styles in the html why not put them in the style bit or is it possible to have a separate style sheet

File: client/test/extension.test.ts [hunk 3]
```diff
@@ -283,10 +296,692 @@ describe('Extension Tests', () => {
...
+                webview: {
+                    onDidReceiveMessage: mySandBox.stub().yields({command: 'openSample', repoName: 'repo1', sampleName: 'sample1'}),
+                    html: ''
+                },
+                onDidDispose: mySandBox.stub()
+            });
+
+            mySandBox.stub(RepositoryRegistry.prototype, 'get').returns({name: 'repo1', path: 'path'});
+            const getContractSample: sinon.SinonStub = mySandBox.stub(SampleView, 'getSamplePage').resolves('<html>Sample Page</html>');
+            // mySandBox.stub(ejs, 'renderFile').resolves('<html>Sample Page</html>');
```
Comment (caroline-church) [inline client/test/extension.test.ts]: can you delete this line if you don't need it

File: client/test/webview/SampleView.test.ts [hunk 4]
```diff
@@ -0,0 +1,734 @@
...
+
+            repositoryRegistryGetStub.should.have.been.calledOnceWithExactly(repositoryName);
+            pathExistsStub.should.have.been.calledOnceWithExactly('/some/path');
+            shellCdStub.should.have.been.calledOnceWithExactly('/some/path');
+            sendCommandStub.should.have.been.calledOnceWithExactly('git checkout -b master origin/master');
+            openNewProjectStub.should.have.not.have.been.called;
+        });
+
+        it('should handle if the local branch already exists', async () => {
+            const repositoryRegistryGetStub: sinon.SinonStub = mySandBox.stub(RepositoryRegistry.prototype, 'get').returns({name: repositoryName, path: '/some/path'});
```
Comment (caroline-church) [inline client/test/webview/SampleView.test.ts]: could this be put in the beforeEach so you don't have it in each test, plus any other duplicate code

File: client/src/commands/createSmartContractProjectCommand.ts [hunk 5]
```diff
@@ -274,7 +274,7 @@ async function getSmartContractLanguageOptions(): Promise<string[]> {
     return parsedJson.contractLanguages;
 }
 
-async function openNewProject(openMethod: string, uri: vscode.Uri): Promise<void> {
+export async function openNewProject(openMethod: string, uri: vscode.Uri): Promise<void> {
```
Comment (caroline-church) [inline client/src/commands/createSmartContractProjectCommand.ts]: This shouldn't be exported if you want to use this function put it somewhere else

**Comment by Jakeeyturner** [comment]
As part of this PR, I'm also bumping the version to v0.2.0

**Pull request merged**